### PR TITLE
Remove action sidebar and bring migrated extension icons to the top nav

### PR DESCRIPTION
**Pull request by philipp-spiess**
This PR removes the action sidebar that was previously used for extensions to significantly reduce the UI clutter on the blob view.

Note: Because we still support `enableLegacyExtensions`, I didn't delete code for the sidebar yet. However all of the code is lazy loaded so it should not impact bundle size. We should do the deletion in a separate PR so it's save to revert.

## Test plan

https://user-images.githubusercontent.com/458591/211868649-f45981de-e754-4381-a93e-bcc921edaf3f.mov

### Testing with `enableLegacyExtensions` on

https://user-images.githubusercontent.com/458591/211868761-378460e4-d36a-4ae1-8886-03dea26a6313.mov

## App preview:

- [Web](https://sg-web-ps-remove-actions-bar.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.07 kb) | -0.02% (-2.55 kb) | -0.02% (-2.61 kb) | 0.42% (+3) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/remove-actions-bar/compare-report.html;tab=live_object) for a full comparison between the commits 7cfe872 and 03acd94 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by philipp-spiess** [comment]
I'll try to see if I can find fitting icons that are monochrome and have less details for the top bar so that the UI is more consistent. We have a bunch of different editor icons though so I’m a bit worried about that 😅

File: client/web/src/open-in-editor/OpenInEditorActionItem.tsx [hunk 1]
```diff
@@ -96,26 +114,36 @@ export const OpenInEditorActionItem: React.FunctionComponent<OpenInEditorActionI
...
                     )
             )}
         </>
-    ) : (
+    ) : // We can not render the editor popover inside the dropdown view yet.
+    // Since the dropdown view is only used on very limited viewport dimensions,
+    // we're okay with having the user go to the settings to configure this
+    // instead.
+    //
+    // Chances are that they won't have an IDE configured on this devices
```
Comment (vdavid) [inline client/web/src/open-in-editor/OpenInEditorActionItem.tsx:128]:
```suggestion
    // Chances are that they won't have an IDE configured on these devices
```

**Review by vdavid — APPROVED** [review]
LGTM, left a typo fix

**Comment by vdavid** [comment]
> I'll try to see if I can find fitting icons that are monochrome and have less details for the top bar so that the UI is more consistent. We have a bunch of different editor icons though so I’m a bit worried about that 😅

That'd be a nice touch, but I'd leave it for a designer, as it's 95% design work and 5% replacing the files and creating a PR. Also, it seems low prio, and I wouldn't bother for now.

**Comment by philipp-spiess** [comment]
> That'd be a nice touch, but I'd leave it for a designer, as it's 95% design work and 5% replacing the files and creating a PR. Also, it seems low prio, and I wouldn't bother for now.

Hm maybe, but:

1. The current way of loading these images via the `` tag should probably changed as well since it causes the images to be loaded last on the page which creates a noticeable lag 

https://user-images.githubusercontent.com/458591/212035587-8603cce2-d5e9-417f-81e5-0893d05b32c0.mov


2. We don't really have enough design resources to help with this at the moment :/

**Comment by philipp-spiess** [comment]
I've changed the icon to "set up your IDE" only for now since this had the most color variance and is also visible to anyone that has not configured anything. It'll now load faster (since it's bundled in the JS) and look more coherent. 

## Before

Screenshot 2023-01-12 at 11 00 26

## After

Screenshot 2023-01-12 at 11 03 34

---

# Open question

Screenshot 2023-01-12 at 11 10 16

I’m unsure what to do about the IDE icons. There are quite a few we support ([see complete list](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@ps/remove-actions-bar/-/blob/client/web/src/open-in-editor/editors.ts?L8-72)) plus especially the JetBrains icons have no low-resolution version: https://www.jetbrains.com/company/brand/#logos-and-icons. The MDI icon set only really has an icon for VS Code (which we could use but then there's inconsistencies if you have another IDE configured at the same time. 

I think we should probably not change the editor icons _just yet_ and fix this up later. 

cc @sourcegraph/design @sourcegraph/code-exploration

**Review by danielmarquespt — APPROVED** [review]

File: client/web/src/open-in-editor/OpenInEditorActionItem.tsx [hunk 2]
```diff
@@ -96,26 +114,58 @@ export const OpenInEditorActionItem: React.FunctionComponent<OpenInEditorActionI
...
+                                <rect x="3" y="16" width="6" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="9" y="9" width="6" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="10" y="2" width="4" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="6" y="12.5" width="6" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="13" y="12.5" width="4" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="16" y="9" width="3" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="6" y="5.5" width="6" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="13" y="5.5" width="3" height="1.5" rx="0.75" fill="currentColor" />
+                                <rect x="10" y="16" width="2" height="1.5" rx="0.75" fill="currentColor" />
+                            </svg>
```
Comment (taras-yemets) [inline client/web/src/open-in-editor/OpenInEditorActionItem.tsx:156]: What do you think about moving this SVG to a separate icon component? It seems to be the approach we [use across the codebase](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph$++%3Csvg+lang:TypeScript&patternType=standard&sm=1&groupBy=repo).

**Review by taras-yemets — APPROVED** [review]
Nice improvement!

**Comment by felixfbecker** [comment]
I'd suggest that we change the current "open in editor" icon (when no editor is set). The current icon shows _code_, which doesn't really explain that it's to open it in your _editor_. Sourcegraph and GitHub all equally show _code_.

I'd suggest `application-edit-outline` from MDI, which shows a desktop application with a pencil to indicate that it's an editor.

CleanShot 2023-01-18 at 09 53 14@2x

I think there is an argument for keeping the actual editor icons as they are, given they are brand logos (although it's a bit inconsistent that they are in color).

But I think we also should change the Blame icon, because currently it only signifies _git_, which could mean a number of things that are not _blame_ (also, what does that mean on a Perforce repo?). It also indicates the active/disabled state only through color, which is a) not WCAG compliant and b) inconsistent with other buttons like raw Markdown mode or line break. Maybe one of these could work because they signify that it's about the authors that have touched the file:

CleanShot 2023-01-18 at 09 57 36@2x 
CleanShot 2023-01-18 at 09 56 59@2x
CleanShot 2023-01-18 at 09 59 46@2x

(`account-details`, `account-group`, `account-edit`)

**Comment by philipp-spiess** [comment]
Yeah agree regarding the open-in-editor setup icon. 

For git blame I think this one would work well. I agree regarding the color not be enough so I took a look at what git lense is doing in vscode and they have one icon that has an outlined and a filled style. Turns out the one that I also think makes the most sense from your suggestions (`account-details` since it also shows lines) has two variants so I'd say ship it? (Will push a PR tomorrow morning)

Screenshot 2023-01-18 at 19 05 44

**Pull request merged**